mlflow/tracking/_tracking_service/client.py

```diff
@@ -884,6 +884,9 @@ def set_terminated(self, run_id, status=None, end_time=None):
         """
         end_time = end_time if end_time else get_current_time_millis()
         status = status if status else RunStatus.to_string(RunStatus.FINISHED)
+        # Tell the store to stop async logging: stop accepting new data and log already enqueued
+        # data in the background. This call is non-blocking.
```

Comment:
>  This call is non-blocking.

Let me confirm my understanding. This call does not block the logging workers to complete, but still wait for the logging thread to drain remained items (L68). Since the thread always wait the configured buffering time (`MLFLOW_ASYNC_LOGGING_WAITING_TIME `) before draining from the queue (L150), this call seems to block for that time, too? If my understanding is correct, I wonder we want to either (1) add note that this will wait for that time (2) update the logic not to wait for the last drain.

Reply:
nice catch! your understanding is correct